http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/TestProxy.scala

```diff
@@ -52,11 +56,21 @@ trait TestProxy { this: ScalaFutures =>
     sslContext.init(keyManagerFactory.getKeyManagers, tmf.getTrustManagers, new SecureRandom)
     val https: HttpsConnectionContext = ConnectionContext.https(sslContext)
 
-    serverBinding = Http().bindAndHandle(backendRoute, "0.0.0.0", proxyConfig.jupyterPort, https).futureValue
+    serverBinding = Http().bindAndHandle(backendRoute, "0.0.0.0", proxyConfig.jupyterPort, https)
+    //Ensures the server starts
+    serverBinding.futureValue
   }
 
   def shutdownProxyServer() = {
-    serverBinding.unbind().futureValue
+    val onceAllConnectionsTerminated: Future[Http.HttpTerminated] =
+      Await.result(serverBinding, 10.seconds)
+        .terminate(hardDeadline = 3.seconds)
+
+    // once all connections are terminated,
+    // - you can invoke coordinated shutdown to tear down the rest of the system:
+    onceAllConnectionsTerminated.flatMap { _ =>
+      system.terminate()
```

Comment:
👍 on terminating `system`...but is `Future` really needed?

Reply:
this is how the documentation recommended doing it, not sure the advantage though